avocado/core/remote/runner.py

```diff
@@ -264,6 +269,10 @@ def run_suite(self, test_suite, variants, timeout=0, replay_map=None):
             self.result.end_tests()
             self.job._result_events_dispatcher.map_method('post_tests',
                                                           self.job)
+            # compare the major version of remote and local version of avocado. If different write to UI and job.log
+            if str(remote_version[0]) != version.MAJOR:
+                APP_LOG.warn(warning_message, '.'.join(map(str, remote_version)), version.VERSION)
+                TEST_LOG.warn(warning_message, '.'.join(map(str, remote_version)), version.VERSION)
```

Comment:
Hmm what troubles me here is the double `join`/`map`/attribution. You can create the `warning_message` inside this `if`,  but there's no need to create it twice from scratch. Do you mind in fixing this? You can force-update then so we will be ready to merge.